source/blog/2015-06-18-ember-data-1-13-released.md

```diff
@@ -0,0 +1,578 @@
+
+In Ember-Data 1.13 we introduced a new adapter hook:
+`handleResponse`. At the same time we deprecating `ajaxSuccess` and
+`ajaxError`.
+
+There been quite a bit of discussion on what an error in the context
+of an adapter actually mean and how much control should we allow to
+the application on deciding given a particular response. At some point
+we made changes to support returning `InvalidError` from
+`ajaxSuccess`. This is why today we replacing both hooks
+(`ajaxSuccess` and `ajaxError`) with just one (`handleResponse`). The
+default implementation will use a couple of new hooks (`isSuccess` and
+`isInvalid`) to decide if a response is a success or an error of
+certain type. It allows for more control over how your application
+handles errors. Default implementation of `isInvalid` is to return
+true if http status code is 422, but you may desside on some other
+semantics, for example check `code` property on JSON-API error object.
+
+We also wanted for quite some time to abstract `jQuery.ajax`
+semantics. We used to reject from the adapter with `jqXHR` object and
+leave the application to deal with it. This strategy have a few
+drawbacks. It made life hard for application authors by making switch
+from one adapter to another a lot of work. It also made life of
+adapter authors hard by failing to provide them a common set of error
+types that can be shared across different backends. A `TimeoutError`
+for example can express a common type of error across a wide range of
+backends. New hooks take three arguments: status code, a hash of
+response headers and parsed payload. It makes them agnostic about
+underlying implementation and will allow to easily use `fetch` in the
+future.
+
+We made a decision in Ember-Data to use JSON-API format as internal
+representation of the data. Naturally, we decided to do the same for
+errors. JSON-API have specified an
+[error objects](http://jsonapi.org/format/#error-objects)
+format. Starting with Ember-Data 1.13 we are using JSON-API format to
+communicate errors from the adapter to the store. Also RESTAdapter and
+JSONAPISerializer supports out of the box JSON-API errors objects in
+the error response payload. It means that the Ruby on Rails inspired
+format we were using as payload for `InvalidError` is replaced with
+proper JSON-API objects. Old format is still supported for now with a
+deprecation warning.
+
+Deprecated format:
+
+```js
+new DS.InvalidError({
+  first_name: [‘is invalid’]
+});
+```
+
+New format:
+
+```js
+new DS.InvalidError([
+  {
+    source: { pointer: ‘data/attributes/first_name’ },
+    details: ‘is invalid’
+  }
+]);
+```
+
+Thanks to [@tchak](https://github.com/tchak) and
+[@twokul](https://github.com/twokul) for working on the design and
+implementation of the new API.
+
+## Significant Deprecations
+
+In addition to new features, Ember Data 1.13 will introduce deprecations for some features and behavior that will be removed in Ember Data 2.0.
```

Comment:
`Data 1.13 will introduce deprecations for some features` -> `Data 1.13 introduces deprecations for features`